### PR TITLE
added coroutine test wrapper

### LINKED ISSUE #101: [soa tests] Get the soa based tests to all run sequentially instead of running them manually one by one

### LINKED ISSUE #136: SOA_bot test running infinitely
I have found this problem both on TACC and a local Docker container

### LINKED ISSUE #138: Set versions numbers for all dependencies

### LINKED ISSUE #141: Remove `gen.coroutine` wrappers from tests
See https://github.com/ALLAN-DIP/baseline_bots/pull/140

### PR DESCRIPTION
**Pull request by kartik2112**
Resolves #141 
Resolves #136 
Resolves #101 
Resolves #138

### CODE REVIEW

File: tests/bot_tests/SOA_test.py [hunk 1]
```diff
@@ -239,7 +239,7 @@ def ally_move_filter(self):
         print("finish test ally move filter")
         stop_io_loop()
 
-    @gen.coroutine
+    @testing.gen_test
     def arse_proposals(self):
```
Comment (trigaten) [inline tests/bot_tests/SOA_test.py:243]: "arse" :)

File: tests/bot_tests/SOA_test.py [hunk 2]
```diff
@@ -449,12 +449,12 @@ def gen_pos_stance_messages(self):
...
-@gen.coroutine
-def main():
-    """Plays a local game with 7 bots"""
-    game = Game()
-    player = SmartOrderAccepterBot("FRANCE", game)
-    f = open("demofile2.txt", "a")
-    f.write("Now the file has more content!")
-    f.close()
-    stop_io_loop()
+# @gen.coroutine
```
Comment (trigaten) [inline tests/bot_tests/SOA_test.py:452]: Delete commented out code

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 3]
```diff
@@ -161,7 +176,8 @@ async def gen_proposal_reply(
                         )
                     )
                 messages.add_message(proposer, str(msg))
-                await self.send_message(proposer, str(msg))
+                if not (self.test_mode):
```
Comment (trigaten) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:179]: what does test mode do? is this documented somewhere?

**Comment by trigaten** [comment]
Looking good, happy to see everything passing CI; what is test mode?

**Comment by kartik2112** [comment]
Added the description for test_mode here: https://github.com/ALLAN-DIP/baseline_bots/blob/soa_test_fix/src/baseline_bots/bots/smart_order_accepter_bot.py#L57

game.send_message is an async function. This was causing issues. So, I'm just bypassing the execution of this command to test the functionality.

File: src/baseline_bots/bots/smart_order_accepter_bot.py (hunk 3 above)
Comment (kartik2112) [reply src/baseline_bots/bots/smart_order_accepter_bot.py:179]: Documented in ini

**Comment by trigaten** [comment]
> Added the description for test_mode here: https://github.com/ALLAN-DIP/baseline_bots/blob/soa_test_fix/src/baseline_bots/bots/smart_order_accepter_bot.py#L57
> 
> game.send_message is an async function. This was causing issues. So, I'm just bypassing the execution of this command to test the functionality.

So this method is not being tested...

What issues was it causing and why can't they be resolved?

**Comment by kartik2112** [comment]
Could someone help me resolve this error in testing? This is wrt using TACC playground and joining the game using multiple powers. I have always launched multiple processes to join the game and start game play. Since this is just a single thread I believe we are getting this error. Creating multiple processes will be an overhaul for such a task IMO. So, if any of you is not able to figure out this error, I would vote against using a network game for code testing and just merge the current set of tests that exclude send_game_message which we have seen in the past that it works.

**Comment by kartik2112** [comment]
I've added a new test case to test `send_message` function of SOA bot. I've kept the rest as it is. This is because having multiple processes to test a simple functionality would have been non-trivial. Thus, simply by keeping 1 player in the game, we can see how the test functionality is working.

**Pull request merged**